integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/scaletest/ScaleTest.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tests.scaletest
+
+import scala.concurrent.duration.NANOSECONDS
+
+import scopt.OptionParser
+
+import org.apache.spark.sql.SparkSession
+
+/**
+ * Entry point for Scale Test
+ */
+object ScaleTest {
+
+  // case class represents all input commandline arguments
+  case class Config(scaleFactor: Int = 1,
+      complexity: Int = 1,
```

Comment:
I would prefer to have a way to specify the query(s) to run and the number of iterations to run that way we can decide what is the best way to run them.

Reply:
I can think of the following way to achieve this: 
an optional int argument `--iterations` to specify the number of rounds a query needs to run by default. If not specified, 1 round.

an optional string argument `--queries`. The string format needs to be 
```
--queries q1:2,q2:3,q3:4,q10:2
``` 
where all entries are comma separated, and each entry are in form `<query-name>:<iterations-to-run>`.

The behavior strategy is: 
1. when `--queries` are not specified, all queries will be run for `--iterations` rounds.
2. when `--queries` are specified, only those specified queries will be run and each of them will run their own iterations.
3. if both of them are specified, the queries that are not specified in `queries` will run `--iterations` rounds while the ones that are specified with iterations will overwrite the default.

What do you think?